src/Plugin.php

```diff
@@ -65,6 +65,28 @@ public static function init() {
 
     add_shortcode('user-login-form', __NAMESPACE__ . '\UserLoginForm::getOutput');
 
+    // Disable self pingbacks.
+    add_action('pre_ping', __CLASS__ . '::pre_ping');
+
+    // Disable pingback functionality and remove x-pingback HTTP header.
+    add_filter('xmlrpc_methods', __CLASS__ . '::xmlrpc_methods');
+    add_filter('wp_headers', __CLASS__ . '::wp_headers');
+
+    // Disables admin discussion options for pingbacks.
+    add_filter('pre_update_default_ping_status', '__return_false');
```

Comment:
Normally it is sufficient to filter the option value; the update is not really relevant.